src/biotite/sequence/io/genbank/metadata.py

```diff
@@ -64,12 +64,45 @@ def get_locus(gb_file):
     # 'LOCUS' field has only one line
     locus_info = lines[0]
 
-    name = locus_info[:18].strip()
-    length = int(locus_info[18 : 28])
-    mol_type = locus_info[32 : 42].strip()
-    is_circular = True if locus_info[43 : 51] == "circular" else False
-    division = locus_info[52 : 55].strip()
-    date = locus_info[56 : 67]
+    fields = str(locus_info).split()
+
+    # The first field will always be the ID
+    name = fields[0]
+
+    # The second field will always be the length followed 
+    # by units (eg 1224 aa)
+    length = int(fields[1])
+
+    # The third field *should* be the molecular type 
+    # but sometimes this is missing
+    # NOTE: remember that fields[2] is the unit for length, 
+    #       eg bp or aa, so we move to fields[3] here.
+    if fields[3] not in ('linear', 'circular'):
+        mol_type = fields[3]
+        next_idx = 4
+    else:
+        mol_type = ''
+        next_idx = 3
+
+
+    # The next field should be the token 'linear' or 'circular', 
+    # but sometimes this is missing
```

Comment:
note: i don't handle the edge case where both mol_type and is_circular are missing because i haven't yet seen that edge case. i can look into to implementing this, though, if it seems worthwhile

Reply:
Added. This was a little tricky for me because I couldn't find a set list of molecular types to check against. I ended up checking against a published list of divisions (see https://www.ncbi.nlm.nih.gov/genbank/samplerecord/#GenBankDivisionA) instead.

Even that didn't quite work, though, because the list they publish in the sample record documentation was actually missing a division, CON, which was fortunately covered in an existing unit test. I checked the full list of divisions currently supported by their search functionality [here](https://www.ncbi.nlm.nih.gov/nuccore/advanced) by selecting Division and clicking "Show index list" to make sure that the list I coded up here is complete.

So long story short, I'm checking against a list of divisions currently supported by GenBank but I'm worried that list may change again in the future making this code somewhat brittle.

Reply:
I agree, but this is still our best effort, so in my opinion it is OK for now. In the future it is easy to add further divisions to the list